File: Python3/1001 - 1200/1037 Valid Boomerang.py
# 28ms, 79.19%; 12.7MB, 100.00%
class Solution:
    def isBoomerang(self, points: List[List[int]]) -> bool:
        k1 = (points[0][0] - points[1][0], points[0][1] - points[1][1])
        k2 = (points[0][0] - points[2][0], points[0][1] - points[2][1])
        return k1[0] * k2[1] != k1[1] * k2[0]

File: 1001 - 1200/1037. Valid Boomerang.cc
// 4ms, 64.84%; 8.3MB, 100.00%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isBoomerang(const vector<vector<int>> &points) {
        int x1 = points[0][0] - points[1][0], y1 = points[0][1] - points[1][1];
        int x2 = points[0][0] - points[2][0], y2 = points[0][1] - points[2][1];
        return x1 * y2 != x2 * y1;
    }
};
